src/buildstream/utils.py

```diff
@@ -1260,14 +1254,7 @@ def _call(*popenargs, terminate=False, **kwargs):
 
     process = None
 
-    old_preexec_fn = kwargs.get("preexec_fn")
```

Comment:
@juergbi I'm not 100% sure if this should logically be like this or `kwargs.setdefault("umask"], stat.S_IWGRP | stat.S_IWOTH)`. Is it expected caller can override umask?

Reply:
I think with original functionality they probably could as caller preexec was called *after* the preexec to set umask.

Reply:
We don't allow plugins to set `preexec_fn` or `umask`, so it wouldn't make a difference right now. However, the point of this umask is to make sure that the umask is deterministic. It's not completely unreasonable to allow plugins to choose a different umask. So I think changing it to `setdefault()` would make sense.